### DISCUSSION TITLE
Lack of superclass function hints

### DISCUSSION THREAD

**Discussion by choboy46** [godotengine/godot-proposals#11408] [Scripting]
Let's say you have a custom class called "Enemy" with a function "get_hit(value)".
```
extends Node2D

class_name Enemy

var hp: int = 0
var maxhp: int = 0

func get_hit(value) -> void:
    hp -= value
```
You create another class named "Spider" which inherits "Enemy" and overrides the "get_hit" function.
```
extends Enemy

class_name Spider

func get_hit(value) -> void:
    hp -= value * 2
```
The problem that I frequently face here is that there's no "hint" offering to autofill a function name from the class that the derived class is inheriting, meaning that if I forgot the exact name of the base class's function, I'll have to go back and check that class's script for the function, so it's a bit inconvenient in that manner.

**Comment by Jesusemora** [comment]
It would be nice if there was autocomplete for overriden functions, and custom overriden functions, normal functions, and overriden godot functions (like `_ready`) should have different colors. Maybe different tones of the current highlight color for custom overriden functions and "Godot" functions.
